enginetest/server_engine.go

```diff
@@ -159,9 +161,15 @@ func (s *ServerQueryEngine) QueryWithBindings(ctx *sql.Context, query string, pa
 		}
 	}
 
+	q := strings.TrimSpace(query)
+	// trim spaces and empty statements
+	q = strings.TrimRightFunc(q, func(r rune) bool {
+		return r == ';' || unicode.IsSpace(r)
+	})
+
```

Comment:
Added this here because when we use the given query to run as prepared, it caches the query with the `;` at the end. This causes inconsistent behavior of storing the `CREATE` statements such as `CREATE VIEW` with `;` at the end, which causes result checking to fail as non-prepared path removes the `;` at the end before running the query.
A thought: maybe we can enforce removing the delimiter character at the end of the query before it gets cached for prepareds? 🤔

Reply:
Good thought – yeah, it seems like the engine taking care of this would be a little cleaner.